core/src/test/scala/org/apache/spark/sql/delta/ShowTableColumnsSuite.scala

```diff
@@ -0,0 +1,229 @@
+        outputColumnNames
+      )
+    }
+  }
+
+  test("delta table: path") {
+    showDeltaColumnsTest(f => s"'${f.toString}'", "", "delta")
+  }
+
+  // when no schema name provided, default schema name is `default`.
+  test("delta table: table identifier") {
+    showDeltaColumnsTest(f => s"delta.`${f.toString}`", "", "delta")
+  }
+
+  test("delta table: table name with separated schema name") {
+    showDeltaColumnsTest(f => s"`${f.toString}`", "delta", "delta")
+  }
+
+  test("non-delta table: path") {
+    // Non-Delta table represent by path (e.g.: '/path/to/table') is NOT supported in
+    // SHOW COLUMNS command.
+    val e = intercept[DeltaFileNotFoundException] {
+      showDeltaColumnsTest(f => s"'${f.toString}'", "", "parquet")
+    }
+    assert(e.getMessage.contains("File path "))
+  }
+
+  test("non-delta table: table identifier with catalog table") {
+    // Non-Delta table represent by catalog identifier (e.g.: sales.line_ite) is supported in
+    // SHOW COLUMNS command.
+    List("json", "csv", "orc", "parquet").foreach { x =>
+      showNonDeltaColumnsTest(x)
+    }
+  }
+
+  test("delta table: path not found") {
+    showDeltaColumnsTest(f => s"'${f.toString}'", "", "delta")
+    val fakeFilePath = s"/invalid/path/to/table"
+    val e = intercept[AnalysisException] {
+      sql(s"SHOW COLUMNS IN `$fakeFilePath`")
+    }
+    assert(
+      e.getMessage()
+        .contains(s"Table or view '$fakeFilePath' not found")
+    )
+  }
+
+  test("delta table: table name not found") {
+    val fakeTableName = s"test_table"
+    val schemaName = s"delta"
+    showDeltaColumnsTest(f => s"$schemaName.`${f.toString}`", "", "delta")
+    val e = intercept[AnalysisException] {
+      sql(s"SHOW COLUMNS IN `$fakeTableName` IN $schemaName")
+    }
+    assert(
+      e.getMessage()
+        .contains(
+          s"Database '$schemaName' not found"
+        )
+    )
+  }
+
+  test("delta table: duplicated schema name") {
+    // If there are 2 schema name in the command, the one attached on table name will be ignored.
+    // e.g.: `SHOW COLUMNS delta.test_table IN epsilon` == `SHOW COLUMNS test_table IN epsilon`.
+    val schemaName = s"epsilon"
+    withDatabase(schemaName) {
+      sql(s"CREATE DATABASE $schemaName")
+      val tableName = "test_table"
+      withTable(tableName) {
+        sql(s"""
+             |CREATE TABLE $schemaName.$tableName(column1 INT, column2 INT)
+             |USING delta
+          """.stripMargin)
+        checkResult(
+          sql(s"SHOW COLUMNS IN delta.$tableName IN $schemaName"),
+          outputColumnValues,
+          outputColumnNames
+        )
+      }
+    }
+  }
+
+  test("delta table: should not use schema name together with path") {
+    // e.g.: SHOW COLUMNS '/path/to/table' IN `delta`
+    // is not allowed.
+    val e = intercept[AnalysisException] {
+      showDeltaColumnsTest(f => s"'${f.toString}'", "delta", "delta")
+    }
+    assert(e.getMessage().contains(s"extraneous input"))
```

Comment:
is this the error from Catalog? If so is it same as the Apache Spark?

Reply:
This error is from catalog. And all test related to path is deleted.